sklearnex/spmd/basic_statistics/tests/test_incremental_basic_statistics_spmd.py

```diff
@@ -0,0 +1,303 @@
+            )
+            incbs_spmd.partial_fit(local_dpt_data, sample_weight=local_dpt_weights)
+        else:
+            incbs_spmd.partial_fit(local_dpt_data)
+
+    if weighted:
+        incbs.fit(dpt_data, sample_weight=dpt_weights)
+    else:
+        incbs.fit(dpt_data)
+
+    assert_allclose(getattr(incbs_spmd, option), getattr(incbs, option))
+
+
+@pytest.mark.skipif(
+    not _mpi_libs_and_gpu_available,
+    reason="GPU device and MPI libs required for test",
+)
+@pytest.mark.parametrize(
+    "dataframe,queue",
+    get_dataframes_and_queues(dataframe_filter_="dpnp,dpctl", device_filter_="gpu"),
+)
+@pytest.mark.parametrize("num_blocks", [1, 2])
+@pytest.mark.parametrize("weighted", [True, False])
+@pytest.mark.parametrize("n_samples", [100, 10000])
+@pytest.mark.parametrize("n_features", [10, 100])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+@pytest.mark.mpi
+def test_incremental_basic_statistics_partial_fit_spmd_synthetic(
+    dataframe, queue, num_blocks, weighted, n_samples, n_features, dtype
+):
+    # Import spmd and batch algo
+    from sklearnex.basic_statistics import IncrementalBasicStatistics
+    from sklearnex.spmd.basic_statistics import (
+        IncrementalBasicStatistics as IncrementalBasicStatistics_SPMD,
+    )
+
+    tol = 2e-3 if dtype == np.float32 else 1e-7
+
+    # Create gold data and process into dpt
+    data = _generate_statistic_data(n_samples, n_features, dtype=dtype)
+    local_data = _get_local_tensor(data)
+    split_local_data = np.array_split(local_data, num_blocks)
+    split_data = np.array_split(data, num_blocks)
+
+    if weighted:
+        weights = _generate_weights(n_samples, dtype=dtype)
+        local_weights = _get_local_tensor(weights)
+        split_local_weights = np.array_split(local_weights, num_blocks)
+        split_weights = np.array_split(weights, num_blocks)
+
+    incbs_spmd = IncrementalBasicStatistics_SPMD()
+    incbs = IncrementalBasicStatistics()
+
+    for i in range(num_blocks):
+        local_dpt_data = _convert_to_dataframe(
+            split_local_data[i], sycl_queue=queue, target_df=dataframe
+        )
+        dpt_data = _convert_to_dataframe(
+            split_data[i], sycl_queue=queue, target_df=dataframe
+        )
+        if weighted:
+            local_dpt_weights = _convert_to_dataframe(
+                split_local_weights[i], sycl_queue=queue, target_df=dataframe
+            )
+            dpt_weights = _convert_to_dataframe(
+                split_weights[i], sycl_queue=queue, target_df=dataframe
+            )
+            incbs_spmd.partial_fit(local_dpt_data, sample_weight=local_dpt_weights)
+            incbs.partial_fit(dpt_data, sample_weight=dpt_weights)
+        else:
+            incbs_spmd.partial_fit(local_dpt_data)
+            incbs.partial_fit(dpt_data)
+
+    for option in options_and_tests:
+        assert_allclose(
+            getattr(incbs_spmd, option[0]), getattr(incbs, option[0]), atol=tol
+        )
```

Comment:
```suggestion
    for option, _, _ in options_and_tests:
        assert_allclose(
            getattr(incbs_spmd, option), getattr(incbs, option), atol=tol
        )
```

for consistency with line 216